cpp/include/raft/matrix/shift.hpp

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
```

Comment:
RAFT is header-only, so it can quickly get confusing for users which file they need to include. We use the general convention of adding "_types" to the file which contain vocabulary types (and are safe to include in downstream APIs, because they, in general, don't require CUDA or any other complex types). Please rename this file to `shift_types.hpp`.